conjure-codegen/src/http_paths.rs

```diff
@@ -0,0 +1,25 @@
+pub enum PathSegment<'a> {
+    Literal(&'a str),
+    Parameter {
+        name: &'a str,
+        regex: Option<&'a str>,
```

Comment:
This is a sneaky hidden feature in Conjure - I think the only variant we expect to actually see in practice is a regex of `.*` on the last segment to allow matches of multi-segment URLs.